### PR TITLE
Finished CSP Agent

### PR DESCRIPTION
**Pull request by gracekumagai**
CSP Agent:
- Stores info about first op and second op's hands
- Keep tracks of interactions between opponents
- If first op and second op and self does not have a card - must be in casefile
- Makes guesses following structure of FC

### CODE REVIEW

File: CSPAgent.py [hunk 1]
```diff
@@ -1,42 +1,126 @@
...
+		self._update_player(self.second_opponent_hand, self.second_opponent_sets)
+		
+		#Prune assigned values in one hand from the other
+		self._update_each_other()
+
+		#Prune assigned values from the hands from the cur dom of the casefile
+		self._update_case(self.first_opponent_hand)
+		self._update_case(self.second_opponent_hand)
+
+		print("Player {}'s cards: {}".format(self.firstOppName, self.first_opponent_hand.get_cards()))
```
Comment (paulsuk) [inline CSPAgent.py]: maybe remove or comment out these print statements

File: CSPAgent.py [hunk 2]
```diff
@@ -46,30 +130,248 @@ def observe_suggestion(self, suggestion, did_respond):
...
+			hand.add_card(card)
+			#print("ADDED {} so now {}".format(domain, hand.get_cards()))
+
+		else:
+			sets.append(domain)
 
-if __name__ == '__main__':
-	agent = CSPAgent('player1')
\ No newline at end of file
+
```
Comment (paulsuk) [inline CSPAgent.py]: add line here

File: CSPAgent.py (hunk 1 above)
Comment (gracekumagai) [inline CSPAgent.py]: Kinda want to keep them in case something breaks / someone wants to reuse elements (for now!)

Comment (gracekumagai) [inline CSPAgent.py]: Done - left the indicator print stmts

**Comment by gracekumagai** [comment]
ALSO FINISHED PROBABILITY AGENT

**Pull request merged**